### PR TITLE
Add 'Tester' interface

### PR DESCRIPTION
**Pull request by tlake**
Creates a more general `Tester` interface and removes the specific `testing.T` object. This allows for tests that use a different but similar testing object to also use this TFTest framework.

This feature arose from the benchmarking tests created for [Layer0 Issue #232](https://github.com/quintilesims/layer0/issues/232) which make use of the `testing.B` object.

### CODE REVIEW

File: logger.go [hunk 1]
```diff
@@ -1,18 +1,14 @@
 package tftest
 
-import (
-	"testing"
-)
-
 type Logger interface {
```
Comment (zpatrick) [inline logger.go]: Let's change the `Logger` interface to 
```
Log(args ...interface{})
Logf(format string, args ...interface{})
```

That way, callers can just passing in `*testing.T` or `*testing.B`.

In this package, we would replace `TestLogger` with a new type, perhaps `IOLogger`:
```
type IOLogger struct {
    writer io.Writer
}

func NewIOLogger(w io.Writer) *IOLogger {
    ...
}

func (i *IOLogger) Log(...)
func (i *IOLogger) Logf(...)
```

(note that `os.Stdout` satisfies the `io.Writer` interface)

File: test_context.go [hunk 2]
```diff
@@ -1,15 +1,18 @@
 package tftest
 
-import (
-	"testing"
-)
+type Tester interface {
```
Comment (zpatrick) [inline test_context.go]: This doesn't need the two `Log` functions

File: test_context.go [hunk 3]
```diff
@@ -1,15 +1,18 @@
...
 
 type TestContext struct {
 	*Context
-	t *testing.T
+	t Tester
 }
 
-func NewTestContext(t *testing.T, options ...ContextOption) *TestContext {
+func NewTestContext(t Tester, options ...ContextOption) *TestContext {
 	options = append([]ContextOption{Log(NewTestLogger(t))}, options...)
```
Comment (zpatrick) [inline test_context.go]: We should drop this line and let the caller specify which logger to use.

File: logger.go (hunk 1 above)
Comment (tlake) [inline logger.go]: ✅

File: test_context.go (hunk 2 above)
Comment (tlake) [inline test_context.go]: ✅

File: test_context.go (hunk 3 above)
Comment (tlake) [inline test_context.go]: ✅

**Pull request merged**